packages/contractkit/src/kit.ts

```diff
@@ -0,0 +1,68 @@
+import { AddressRegistry } from 'src/address-registry'
+import { Address, CeloToken } from 'src/base'
+import { WrapperCache } from 'src/contract-cache'
+import { sendTransaction, TxOptions } from 'src/utils/send-tx'
+import { toTxResult } from 'src/utils/tx-result'
+import { Web3ContractCache } from 'src/web3-contract-cache'
+import Web3 from 'web3'
+import { TransactionObject, Tx } from 'web3/eth/types'
+
+export function newKit(url: string) {
+  return newKitFromWeb3(new Web3(url))
+}
+
+export function newKitFromWeb3(web3: Web3) {
+  return new ContractKit(web3)
+}
+
+export class ContractKit {
+  readonly registry: AddressRegistry
+  readonly _web3Contracts: Web3ContractCache
+  readonly contracts: WrapperCache
+
+  defaultOptions: TxOptions
+  constructor(readonly web3: Web3) {
+    this.defaultOptions = {
+      gasInflationFactor: 1.3,
+    }
+
+    this.registry = new AddressRegistry(this)
+    this._web3Contracts = new Web3ContractCache(this)
+    this.contracts = new WrapperCache(this)
+  }
+
+  async setGasCurrency(token: CeloToken) {
+    this.defaultOptions.gasCurrency = await this.registry.addressFor(token)
+  }
+
+  set defaultAccount(address: Address) {
+    this.web3.eth.defaultAccount = address
+  }
+
+  get defaultAccount() {
+    return this.web3.eth.defaultAccount
+  }
+
+  setGasCurrencyAddress(address: Address) {
+    this.defaultOptions.gasCurrency = address
+  }
+
+  sendTransaction(tx: Tx) {
+    const promiEvent = this.web3.eth.sendTransaction({
```

Comment:
There's a hack around `gasPrice` that we should do. Basically, `web3` will try to set a default gas price for us, for which it uses `5 * minimum_gas_price_in_gold`, no matter what the `gasCurrency` is.

If we want to suggest a good gas price when we pay in other currencies we need to set `gasPrice: '0'`.  This will prevent `web3` from setting a default gas price and allow `celo-blockchain` to set the gas price, for which it will consider the `gasCurrency`

Reply:
what happens if it's a locally signed tx?
I imagine that we need to set the gasPrice there, since celo-blockchain won't be able to modify the tx

Reply:
> I imagine that we need to set the gasPrice there, since celo-blockchain won't be able to modify the tx

Yes

Reply:
I'll add a comment for now